docs/topics/settings.txt

```diff
@@ -291,6 +291,23 @@ Note that calling ``django.setup()`` is only necessary if your code is truly
 standalone. When invoked by your Web server, or through :doc:`django-admin
 </ref/django-admin>`, Django will handle this for you.
 
+Note also that you shouldn't call django.setup() in any code that's likely to be
+loaded by django itself.  So, in standalone scripts, it's best to put calls to
+django.setup() inside an `if __main__` block, like this:
+
+
+    if __name__ == '__main__':
+        import django
+        django.setup()
+
+Better still, try to keep standlone scripts as minimal as possible and put
+application logic in other modules of your application.  Aside from the problems
+that arise from multiple calls to django.setup(), standalone scripts also usually
+need to make changes to sys.path, and that can lead to multiple copies of the same
+being imported in two different ways, which in turn can lead to all kinds of
+hard-to-debug errors.
```

Comment:
It's less hard to debug since Django shouts at you when the same model is imported through two different paths, though ;-)